scripts/updatekinto.js

```diff
@@ -0,0 +1,62 @@
+"use strict";
+
+const got = require("got");
+
+const SERVER = "https://settings-writer.prod.mozaws.net/v1";
+const CID = "fxmonitor-breaches";
+const DEBUG = true;
+
+async function run() {
+  const RemoteSettingsBreachesSet = new Set((await got(
+    "https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/fxmonitor-breaches/records",
+    { json: true }
+  )).body.data.map(b => b.Name));
+
+  const hibp_breaches = (await got(
+    "https://haveibeenpwned.com/api/v2/breaches",
+    { json: true }
+  )).body;
+
+  const new_breaches = [];
+
+  for (const breach of hibp_breaches) {
+    if (breach.IsSpamList || breach.IsRetired || !breach.IsVerified || !breach.Domain) {
+      continue;
+    }
+
+    if (RemoteSettingsBreachesSet.has(breach.Name)) {
+      continue;
+    }
+
+    new_breaches.push(breach);
+  }
+  for (const breach of new_breaches) {
+    const data = {
+      Name: breach.Name,
+      Domain: breach.Domain,
+      BreachDate: breach.BreachDate,
+      PwnCount: breach.PwnCount,
+      AddedDate: breach.AddedDate,
+    };
+
+    if (DEBUG) {
+      console.log(data);
+      continue;
+    }
+
+    try {
+      await got.post(`${SERVER}/buckets/main-workspace/collections/${CID}/records`, {
+        headers: {
+          "Content-Type": "application/json",
+          "authorization": "Bearer <insert token here>",
+        },
+        body: JSON.stringify({data: data}),
+      });
+    } catch (e) {
+      console.log(e);
+      return;
```

Comment:
Should we do a `process.exit(1)` or `process.exitCode = 1` or something here so we exit with a non-zero status?

Reply:
I guess since we're updating Kinto in a loop here, it would make more sense to use `process.exitCode` otherwise `.exit(1)` will abort after the first error.

Not sure if there is a way to just push all new records up to Kinto in a single request, or if it's worth the effort.

Reply:
There is a batch mode, but there's a limit on how many records you can upload per batch if I remember correctly - I tested that like a month or two ago.

I don't think it's worth the effort, since this will likely run something like every day and I'd expect either 0 or 1 new breach added per run MOST of the time.

Reply:
> Should we do a `process.exit(1)` or `process.exitCode = 1` or something here so we exit with a non-zero status?

Yes indeed, proper exit codes is on my to-do list for this WIP. Thanks!

Reply:
> I don't think it's worth the effort, since this will likely run something like every day and I'd expect either 0 or 1 new breach added per run MOST of the time.

Yeah, it definitely looks like having 2+ breaches per day is rare.
With a daily high of 8 on Nov 7 2016!

```
total breaches: 345

48 days had 2+ breaches added
12 days had 3+ breaches added
 4 days had 4+ breaches added
 2 days had 5+ breaches added

12/27/2016 ==> 4
11/7/2016 ==> 8
 9/5/2016 ==> 4
12/3/2013 ==> 5
```

<details>
<summary>source</summary>

```js
const { getBreaches } = require("monitor-breaches-axios");

async function main(minPerDay = 2) {
  const { breaches } = await getBreaches();
  const sortedBreaches = breaches.sort(
    (breachA, breachB) => breachB.AddedDate - breachA.AddedDate
  );
  const breachDays = {};
  sortedBreaches.forEach(breach => {
    const day = breach.AddedDate.toLocaleDateString();
    const res = breachDays[day] || [];
    breachDays[day] = [...res, breach.Title];
  });

  let count = 0;
  Object.entries(breachDays).forEach(([key, value]) => {
    if (value.length >= minPerDay) {
      console.log(key, "==>", value.length);
      count++;
    }
  });
  console.log(`\n${count} days had ${minPerDay}+ breaches added`);
  console.log(`total breaches: ${breaches.length}`);
}

main(2);
```
</details>

Reply:
Nice, thanks for verifying my claim!